rosetta/services/block.go

```diff
@@ -48,31 +48,34 @@ func (s *BlockAPI) Block(
 		return nil, rosettaError
 	}
 
-	// Format genesis block if it is requested.
-	if blk.Number().Uint64() == 0 {
-		return s.genesisBlock(ctx, request, blk)
-	}
-
 	currBlockID = &types.BlockIdentifier{
 		Index: blk.Number().Int64(),
 		Hash:  blk.Hash().String(),
 	}
-	prevBlock, err := s.hmy.BlockByNumber(ctx, rpc.BlockNumber(blk.Number().Int64()-1).EthBlockNumber())
-	if err != nil {
-		return nil, common.NewError(common.CatchAllError, map[string]interface{}{
-			"message": err.Error(),
-		})
-	}
-	prevBlockID = &types.BlockIdentifier{
-		Index: prevBlock.Number().Int64(),
-		Hash:  prevBlock.Hash().String(),
+
+	if blk.NumberU64() == 0 {
```

Comment:
please add a comment here on the special handling.

Reply:
IMO comment here would not serve any benefit. The genesis block handling is clear by the nature of the code.